compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -2253,7 +2245,8 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         previous_stack: TraitObligationStackList<'o, 'tcx>,
         obligation: &'o TraitObligation<'tcx>,
     ) -> TraitObligationStack<'o, 'tcx> {
-        let fresh_trait_pred = obligation.predicate.fold_with(&mut self.freshener);
+        let fresh_trait_pred =
+            obligation.predicate.fold_with(&mut self.infcx.freshener_keep_static());
```

Comment:
keeping `'static` here is wrong. Changing this separately in #102472 which is blocked on #102635